public/js/notifications-worker.js

```diff
@@ -0,0 +1,276 @@
+            // block request as the event-stream unneeded for now (2 tabs are already connected)
+            event.respondWith(new Response(
+                null,
+                {
+                    status: 204,
+                    statusText: 'No Content'
+                }
+            ));
+        }
+    }
+});
+selfSW.addEventListener('notificationclick', (event) => {
+    event.notification.close();
+    if (! ('action' in event)) {
+        self.clients.openWindow(event.notification.data.url).then();
+    } else {
+        switch (event.action) {
+            case 'viewIncident':
+                self.clients.openWindow(event.notification.data.url).then();
+                break;
+            case 'dismiss':
+                break;
+        }
+    }
+});
+
+function processMessage(event) {
+    if (event.data) {
+        let data = JSON.parse(event.data);
+        switch (data.command) {
+            case 'tab_force_reclaim':
+                /*
+                 * trigger the claim process as there seems to be new clients in our scope which aren't under our
+                 * control
+                 */
+                self.clients.claim().then(() => {
+                    self.console.log(_PREFIX + "reclaimed all tabs.");
+                });
+                break;
+            case 'tab_notification':
+                /*
+                 * displays a notification through the service worker (if the permissions have been granted)
+                 */
+                if (('Notification' in self) && (self.Notification.permission === 'granted')) {
+                    const notification = data.notification;
+                    let title = '';
+                    let severity = 'unknown';
+
+                    // match severity
+                    switch (notification.payload.severity) {
+                        case 'ok':
+                            severity = 'ok';
+                            break;
+                        case 'warn':
+                            severity = 'warning';
+                            break;
+                        case 'crit':
+                            severity = 'critical';
+                            break;
+                    }
+
+                    // build title
+                    if (notification.payload.service !== '') {
+                        title += "'" + notification.payload.service + "' on ";
+                    }
+                    title += "'" + notification.payload.host + "'";
+
+                    self.registration.showNotification(
+                        title,
+                        {
+                            icon: _BASE_URL + '/img/notifications/icinga-notifications-' + severity + '.webp',
+                            body: 'changed to severity ' + severity,
+                            data: {
+                                url:
+                                    _BASE_URL
```

Comment:
The `_BASE_URL` is somehow still empty because the `View Incident` does not contain the prefix `icingaweb2`. Which leads to `404` page.